### DISCUSSION TITLE
Allow injection of generation context into evaluation prompt

### DISCUSSION THREAD

**Discussion by dveyarangi** [langfuse/langfuse#3905] [Support] [answered]
### Describe the feature or potential improvement

The idea is to allow injecting previous LLM reply as variable into evaluation prompt, to allow evaluating user reply sentiment, feedback presence, etc, in an extended conversation context. I understand that this is possible via API, but it would be nice to have this in the web ui also.

**Comment by marcklingen** [comment]
thanks for sharing this! conversation context meaning within the same langfuse `session` but outside the current trace?

**Reply by dveyarangi** [reply]
I mean a way to inject previous llm reply (that may be appearing in the context of same generation/trace). And yes, it is the output of previous generation in the same session

**Reply by dveyarangi** [reply]
basically, if you consider a chat flow, with user request and bot messages ping-poing, the user reply may indicate a feedback on the previous llm reply. But to evaluate if this is feedback or not, evaluator need to know what was the bot message that triggered user response

**Reply by marcklingen** [reply, marked as answer]
makes sense! what you could do today is to take the generation input/output and pass it to an llm eval in langfuse while prompting for whether there are patterns in the reaction of the user to previous messages. I understand that being able to query specific messages from the generation object would be useful here. This is tracked here: https://github.com/orgs/langfuse/discussions/3237

**Comment by maxdeichmann** [comment]
@dveyarangi we built and released this yesterday. Please let us know if this works for you.

https://langfuse.com/changelog/2025-02-06-llm-as-a-judge-jsonpath
